iron-fit-behavior.html

```diff
@@ -0,0 +1,190 @@
+    },
+
+    get _fitWidth() {
+      return this.fitInto === window ? this.fitInto.innerWidth : this.fitInto.getBoundingClientRect().width;
+    },
+
+    get _fitHeight() {
+      return this.fitInto === window ? this.fitInto.innerHeight : this.fitInto.getBoundingClientRect().height;
+    },
+
+    attached: function() {
+      if (this.autoFit) {
+        if (getComputedStyle(this).display === 'none') {
+          setTimeout(function() {
+            this.fit();
+          }.bind(this));
+        } else {
+          this.fit();
+        }
+      }
+    },
+
+    fit: function() {
+      this._discoverInfo();
+      this.constrain();
+      this.center();
+    },
+
+    /**
+     * Memoize information needed to position and size the target element.
+     */
+    _discoverInfo: function() {
+      if (this._fitInfo) {
+        return;
+      }
+      var target = getComputedStyle(this);
+      var sizer = getComputedStyle(this._sizingTarget);
+      this._fitInfo = {
+        positionedBy: {
+          vertically: target.top !== 'auto' ? 'top' : (target.bottom !== 'auto' ?
+            'bottom' : null),
+          horizontally: target.left !== 'auto' ? 'left' : (target.right !== 'auto' ?
+            'right' : null),
+          css: target.position
+        },
+        sizedBy: {
+          height: sizer.maxHeight !== 'none',
+          width: sizer.maxWidth !== 'none'
+        },
+        margin: {
+          top: parseInt(target.marginTop) || 0,
+          right: parseInt(target.marginRight) || 0,
+          bottom: parseInt(target.marginBottom) || 0,
+          left: parseInt(target.marginLeft) || 0
+        }
+      };
+    },
+
+    /**
+     * Resets the target element's position and size constraints, and clear
+     * the memoized data.
+     */
+    resetFit: function() {
+      if (!this._fitInfo || !this._fitInfo.sizedBy.height) {
+        this._sizingTarget.style.maxHeight = '';
+        this.style.top = '';
+      }
+      if (!this._fitInfo || !this._fitInfo.sizedBy.width) {
+        this._sizingTarget.style.maxWidth = '';
+        this.style.left = '';
+      }
+      this.style.position = this._fitInfo.css;
+      this._fitInfo = null;
+    },
+
+    refit: function() {
+      this.resetFit();
+      this.fit();
+    },
+
+    constrain: function() {
+      var info = this._fitInfo;
+      // position at (0px, 0px) if not already positioned, so we can measure the natural size.
+      if (!this._fitInfo.positionedBy.vertically) {
+        this.style.top = '0px';
+      }
+      if (!this._fitInfo.positionedBy.horizontally) {
+        this.style.left = '0px';
+      }
+      // need border-box for margin/padding
+      this._sizingTarget.style.boxSizing = 'border-box';
+      // constrain the width and height if not already set
+      var rect = this.getBoundingClientRect();
+      if (!info.sizedBy.height) {
+        this._sizeDimension(rect, info.positionedBy.vertically, 'top', 'bottom', 'Height');
```

Comment:
It seems as though there are only two calls of this private method, and the values of the last three arguments are unique to the value of the second argument. Would it suit the use-case to derive the last three arguments from the second argument within the method?